### ISSUE TITLE
Issue warning if read_html / read_xml passed vector > length 1

### ISSUE THREAD

**Issue by thelatemail** [r-lib/xml2#271]
Source issue here: https://stackoverflow.com/questions/57469174/function-in-a-dplyr-pipeline-executes-the-first-item-repeatedly-rather-than-one
I think the root cause of this is in the *xml2* package. Apologies if I'm putting this in the wrong spot.

When providing `read_html` with a vector input of length > 1, a warning could be issued that only the first string is processed. This would be particularly helpful in using the function inside `dplyr::mutate`, where the same output string is given for each row silently, and possibly unexpectedly.

E.g.:

```r
library(dplyr)
library(tibble)
library(rvest)

mtcars %>%
  rownames_to_column("Car") %>%
  select(Car) %>%
  mutate(html_string = paste0("<a>",Car,"</a>")) %>%
  mutate(cleaned_string = html_text(read_html(html_string)))
```

Tested on R 3.6.1, xml2_1.2.0, Ubuntu 18.04, - as far as I can tell, there is no change in the more recent development version of xml2 that would issue a warning.

**Comment by jimhester** [comment]
We should likely error if `read_html()` is passed a vector of length greater than 1

**Comment by hlydecker** [comment]
Is there any potential to make read_xml / read_xml work with a vector input? It would be nice to be able to use these functions within dplyr::mutate. An example use case would be creating a column with values extracted from reading multiple xmls using read_xml()

**Comment by jimhester** [comment]
Fixed by fbdbfe6
